python_modules/dagster/dagster/_core/execution/context/system.py

```diff
@@ -822,8 +821,14 @@ def step_materializes_assets(self) -> bool:
             )
             return asset_info is not None
 
-    def record_logical_version(self, asset_key: AssetKey, logical_version: LogicalVersion) -> None:
-        self._generated_logical_versions[asset_key] = logical_version
+    def set_logical_version(self, asset_key: AssetKey, logical_version: "LogicalVersion") -> None:
```

Comment:
very strong aversion to making context mutable in this fashion

Reply:
I'm open-minded on other approaches but just want to point out that this PR does not introduce this mutability.